### PR TITLE
Fix env checker single-step-env edge case (#1521)

### PR DESCRIPTION
**Pull request by lutogniew**
## Description
Before this change, env checker failed to `reset()` the tested environment before calling `step()` when checking for `Inf` / `NaN`. This could cause environments which happened to have only one `step()` available before the episode was terminated to fail.

This is now fixed.

## Motivation and Context

Closes [#1521](https://github.com/DLR-RM/stable-baselines3/issues/1521).
- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: tests/test_env_checker.py [hunk 1]
```diff
@@ -112,3 +112,58 @@ def step(self, action):
     test_env = TestEnv()
     with pytest.raises(AssertionError, match=error_message):
         check_env(env=test_env)
+
+
+class StepCalledAfterEnvTerminatedException(Exception):
```
Comment (araffin) [inline tests/test_env_checker.py:117]: I don't think that's needed here, as we are not catching it, an assert will be enough.

File: tests/test_env_checker.py [hunk 2]
```diff
@@ -112,3 +112,58 @@ def step(self, action):
...
+            raise StepCalledAfterEnvTerminatedException
+
+        observation = 0
+        reward = 0.0
+        self._terminated = self._steps_called >= self._steps_before_termination
+        truncated = False
+
+        return observation, reward, self._terminated, truncated, {}
+
+    def render(self, mode: str = "human") -> None:
```
Comment (araffin) [inline tests/test_env_checker.py:158]: minor: mode argument has been removed in gymnasium

```suggestion
    def render(self) -> None:
```

File: tests/test_env_checker.py [hunk 3]
```diff
@@ -112,3 +112,58 @@ def step(self, action):
...
         check_env(env=test_env)
+
+
+class StepCalledAfterEnvTerminatedException(Exception):
+    pass
+
+
+class LimitedStepsTestEnv(gym.Env):
+    metadata = {"render_modes": ["human"]}
+    render_mode = None
```
Comment (araffin) [inline tests/test_env_checker.py:123]: maybe add it as a parameter too?

**Review by araffin — COMMENTED** [review]
Thanks for the PR, overall LGTM =) just some minor comments

File: tests/test_env_checker.py (hunk 3 above)
Comment (araffin) [reply tests/test_env_checker.py:123]: yes, I meant allow `render_mode` as a parameter to the constructor, or remove everything, yes.

**Review by araffin — APPROVED** [review]

**Comment by lutogniew** [comment]
@araffin Thank you for the review and merge!

**Pull request merged**